app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt

```diff
@@ -174,40 +208,69 @@ class RecFragment(contentLayoutId: Int) : Fragment(contentLayoutId) {
         }
     }
 
-    private fun switchRecordState(counterText : MutableState<String>, recordButtonText : MutableState<String>) {
+    private fun switchRecordState(counterText : MutableState<String>,
+                                  recordButtonText : MutableState<String>,
+                                  debugValueText : MutableState<String>,
+                                  thresholdValue : MutableState<String>) {
         if (!isRecording) {
             // Set the timer
             recordTimer = onTickTimer(COUNT_DOWN_INTERVAL) { millisUntilFinished ->
                 counterText.value = ((MILLIS_IN_FUTURE - millisUntilFinished) / Companion.COUNT_DOWN_INTERVAL).toString()
             }
 
+            processSamplesTimer = onTickTimer(NOTE_SAMPLE_INTERVAL){
+                val raw_samples = ShortArray(NOTE_SAMPLE_WINDOW_SIZE)
+                val bytesRead = audioRecorder.read(raw_samples, 0, NOTE_SAMPLE_WINDOW_SIZE)
+                val samples = Signal(NOTE_SAMPLE_WINDOW_SIZE)
+                for (i in 0 until NOTE_SAMPLE_WINDOW_SIZE){
+                    samples[i] = raw_samples[i].toFloat() * (1.0f / 32768.0f)
+                }
+                val debugValue = transcriber.process_samples(samples)
+                debugValueText.value = debugValue.toString()
+            }
             //start recording
             recordButtonText.value = "Stop recording"
             // Start the timer
             recordTimer.start()
-            startRecording()
+            processSamplesTimer.start()
+            startRecording(thresholdValue)
         } else {
             //stop recording
             // Stop the timer
             recordTimer.cancel()
+            processSamplesTimer.cancel()
             // Set the recording time to 0
             counterText.value = "Recording saved!"
             recordButtonText.value = "Start recording"
             stopRecording()
         }
     }
 
-
     fun getOutputFilePath(): String {
         return requireContext().externalCacheDir?.absolutePath + "/recording.3gp"
     }
 
     @SuppressLint("Permissions are checked before calling this method", "MissingPermission")
-    private fun startRecording() {
+    private fun startRecording(thresholdValue: MutableState<String>) {
         // Initialize the AudioRecord
         audioRecorder = AudioRecord(audioSource,
             Companion.SAMPLE_RATE_IN_HZ, channelConfig, audioFormat, bufferSize)
 
+        val threshold = try {
+           thresholdValue.value.toDouble()
+        } catch (_: java.lang.NumberFormatException) {
+            thresholdValue.value = "0.95"
+            0.95
```

Comment:
Maybe it could be nice to not have a magic number here

Reply:
That’s meant to be temporary. It will either be a proper constant or linked to some setting page for the user to set it.